module-2/lab-subsetting-and-descriptive-stats/your-code/main.ipynb

```diff
@@ -21,13 +21,14 @@
   },
```

Comment:
- Your plots (both histogram and boxplot) should work if you successfully installed the matplotlib library. Your challenge 3 hist did work, it is not possible that the same plot works and doesn't work in the same notebook. Let's see tomorrow what the problem is because you will need in the following days. 
- This code in the bonus questions does not work 
```Python
SalDifference = (employee.groupby('Department').apply(lambda x: employee. employee.groupby'Department').min())
```
This is a possible way of solving this :
```Python 
employee.groupby(['Department']).agg({'Salary':lambda x: np.max(x)-np.min(x)})
```
- It's a good practice to skip useless columns when importing data sets. This is the case of challenge 3, you can skip the unnamed column in order to avoid any kind of misunderstanding. 
- Describe function has three parameters that can be defined by user (percentiles=None, include=None, exclude=None), https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.describe.html). You are calling describe with default configuration (None for three parameters) that's because you need to use (). Dtypes is just a property, it doesn't accept any argument. 
- Select all the customers we have in Spain: You must select unique values otherwise as you have a data set of orders and one customer can have more than an order in a country, the result is a list of duplicated id's. 
A way of doing this is using set method: 
```Python
set(orders.loc[orders['Country']=='Spain']['CustomerID'])
```
- How many customers do we have in Spain? Hint : Use value_counts(): one way of doing is calculating the len of the above set
```Python
len(set(orders.loc[orders['Country']=='Spain']['CustomerID']))
```
- What's the most expensive item?: 2 ways of doing it: 
```Python
#Ivan's 
print('Most expensive item: ',orders.loc[orders['UnitPrice']==max(orders['UnitPrice'])]['Description'].values[0])
#Albertoss 
orders_most_expensive = orders[orders['UnitPrice'] == orders['UnitPrice'].max()]
```